test/tools/yulInterpreter/Interpreter.cpp

```diff
@@ -126,28 +126,41 @@ void Interpreter::operator()(ForLoop const& _forLoop)
 	openScope();
 	for (auto const& statement: _forLoop.pre.statements)
 		visit(statement);
+	if (m_state.controlFlowState == ControlFlowState::Leave)
```

Comment:
This may be too late, if the first for loop pre statement is a leave, all subsequent statements will be evaluated even if they shouldn't be.

Reply:
Example:

```
{
function foo_n_0()
{
for {leave invalid()} 1 {} {}
}
foo_n_0()
}
```

evaluates `invalid()` and leaves a trace although it shouldn't

Reply:
Ah right, it's not evaluated as a block! Thanks!

Reply:
Fixed and added test.